src/coprocessor/endpoint.rs

```diff
@@ -425,69 +377,127 @@ impl Drop for RequestTracker {
     }
 }
 
-#[derive(Debug)]
 pub struct RequestTask {
-    req: Request,
-    cop_req: CopRequest,
-    ctx: ReqContext,
+    req_ctx: Arc<ReqContext>,
```

Comment:
Could we remove `Arc`? And I think name `ctx` is ok.

Reply:
If I remembered, the `Arc` here is to eliminate a `clone` introduced in previous PR.. I will prepare a material about why `Arc` is required here for you later.

Reply:
Maybe you could define the `RequestHandleBuilder` as
```
 Box<FnBox(Box<storage::Snapshot + 'static>,ReqContext) -> Result<Box<RequestHandler>> + Send>;
```

So the `Arc` could be removed.